js/ui/text_box/ui.text_editor.mask.strategy.default.js

```diff
@@ -60,48 +55,49 @@ class DefaultMaskStrategy extends BaseMaskStrategy {
         this._inputHandlerTimer = setTimeout((function() {
             this._caret({ start: caret.start, end: caret.start });
 
-            this._maskKeyHandler(event, function() {
-                this._handleKey(char);
-                return true;
-            });
+            this._maskKeyHandler(event, () => this._handleKey(char));
         }).bind(this.editor));
     }
 
     _backspaceHandler(event) {
+        const { editor } = this;
         this._keyPressHandled = true;
 
         const afterBackspaceHandler = (needAdjustCaret, callBack) => {
             if(needAdjustCaret) {
-                this.editor._direction(this.DIRECTION.FORWARD);
-                this.editor._adjustCaret();
+                editor._direction(this.DIRECTION.FORWARD);
+                editor._adjustCaret();
             }
             const currentCaret = this.editorCaret();
-            clearTimeout(this._backspaceHandlerTimeout);
-            this._backspaceHandlerTimeout = setTimeout(function() {
-                callBack(currentCaret);
+
+            return new Promise((resolve) => {
```

Comment:
We need to import the `Promise` polyfill. See: https://github.com/DevExpress/DevExtreme/blob/3f6e5b98c8c9e6192581a2ed56b3e081adb79bc0/js/ui/map/provider.dynamic.js#L1